FWCore/Sources/src/IDGeneratorSourceBase.cc

```diff
@@ -164,12 +164,6 @@ namespace edm {
       BASE::setNewLumi();
       return newFile ? BASE::IsFile : BASE::IsRun;
     }
-    if (BASE::processingMode() == BASE::Runs) {
-      return newFile ? BASE::IsFile : BASE::IsRun;
-    }
-    if (BASE::processingMode() == BASE::RunsAndLumis) {
-      return newFile ? BASE::IsFile : BASE::IsLumi;
-    }
```

Comment:
So the fix is that in case `BASE::IsFile` is `false`, the return value should always be `BASE::IsRun`?

Reply:
The fix was to ignore `BASE::processMode()` since the parent class actually handles that in the proper way. I.e. the parent expects the inheriting modules to still pass it 'IsEvent' calls.